### PR TITLE
chore: migrate from dep to Go modules

### LINKED ISSUE #17: Update git2consul to use Go Modules instead of dep
## What happens?

When developing git2consul, `dep` is used to manage dependencies.

## What were you expecting to happen?

Go Modules should be used instead of dep.

## Any errors, stacktrace, logs? 
N/A

### PR DESCRIPTION
**Pull request by vinny-sabatini**
## Description

This migrates `git2consul` to use Go modules for managing dependencies rather than dep. Go modules was introduced in version 1.11, so building using Go version 1.10 will no longer be supported.

## Type of change

* New feature (non-breaking change which adds functionality)
* This change requires a documentation update

## Environment

* Runtime version(Java, Go, Python, etc): Go 1.13.5

Fixes #17

### CODE REVIEW

**Comment by vinny-sabatini** [comment]
I don't believe Snyk supports Go modules yet.

File: .travis.yml [hunk 1]
```diff
@@ -2,21 +2,21 @@
 language: go
 matrix:
   include:
-    - go: 1.10.x
     - go: 1.11.x
     - go: 1.12.x
```
Comment (seanmalloy) [inline .travis.yml]: Let's remove version 1.11 and 1.12 too.

File: .travis.yml [hunk 2]
```diff
@@ -2,21 +2,21 @@
...
 before_install:
-  - go get github.com/golang/dep/cmd/dep
+  - export GO111MODULE=on
   - go get golang.org/x/lint/golint
 
 install:
   - # skip
 
 script:
+  - go mod vendor
```
Comment (seanmalloy) [inline .travis.yml]: Let's not use `go mod vendor`. We don't need to rely on the `vendor` directory when using go modules. I think you should just be able to delete this line.

File: README.md [hunk 3]
```diff
@@ -201,25 +191,27 @@ Sample config with ssh auth
...
 ### Dependencies
-* Go 1.10+
-* [dep](https://github.com/golang/dep)
+* Go 1.13+
 
 ### Compiling From Source
 ```
-$ dep ensure
+$ export GO111MODULE=on
+$ go mod vendor
```
Comment (seanmalloy) [inline README.md]: Remove `go mod vendor`.

File: README.md [hunk 4]
```diff
@@ -201,25 +191,27 @@ Sample config with ssh auth
...
 $ go build -o build/bin/git2consul
 ```
 
 For Development/Debugging
 ```
+$ export GO111MODULE=on
+$ go mod vendor
```
Comment (seanmalloy) [inline README.md]: Remove `go mod vendor`.

**Comment by vinny-sabatini** [comment]
@seanmalloy good call, I've addressed your suggestions

**Pull request merged**